backend/app/rest/proxy/image.go

```diff
@@ -14,36 +19,77 @@ import (
 	"github.com/pkg/errors"
 
 	"github.com/umputun/remark/backend/app/rest"
+	"github.com/umputun/remark/backend/app/store/image"
 )
 
 // Image extracts image src from comment's html and provides proxy for them
 // this is needed to keep remark42 running behind of HTTPS serve all images via https
 type Image struct {
-	RemarkURL string
-	RoutePath string
-	Enabled   bool
-	Timeout   time.Duration
+	RemarkURL     string
+	RoutePath     string
+	HTTP2HTTPS    bool
+	CacheExternal bool
+	Timeout       time.Duration
+	ImageService  *image.Service
 }
 
-// Convert all img src links without https to proxied links
+// Convert img src links to proxied links depends on enabled options
 func (p Image) Convert(commentHTML string) string {
-	if !p.Enabled || strings.HasPrefix(p.RemarkURL, "http://") {
-		return commentHTML
+	if p.CacheExternal {
+		imgs, err := p.extract(commentHTML, func(img string) bool { return !strings.HasPrefix(img, p.RemarkURL) })
+		if err != nil {
+			return commentHTML
+		}
+		commentHTML = p.replace(commentHTML, imgs)
+	}
+
+	if p.HTTP2HTTPS && !strings.HasPrefix(p.RemarkURL, "http://") {
+		imgs, err := p.extract(commentHTML, func(img string) bool { return strings.HasPrefix(img, "http://") })
+		if err != nil {
+			return commentHTML
+		}
+		commentHTML = p.replace(commentHTML, imgs)
 	}
 
-	imgs, err := p.extract(commentHTML)
+	return commentHTML
+}
+
+// extract gets all images matching predicate and return list of src
+func (p Image) extract(commentHTML string, imgSrcPred func(string) bool) ([]string, error) {
+	doc, err := goquery.NewDocumentFromReader(strings.NewReader(commentHTML))
 	if err != nil {
-		return commentHTML
+		return nil, errors.Wrap(err, "can't create document")
+	}
+	result := []string{}
+	doc.Find("img").Each(func(i int, s *goquery.Selection) {
+		if im, ok := s.Attr("src"); ok {
+			if imgSrcPred(im) {
+				result = append(result, im)
+			}
+		}
+	})
+	return result, nil
+}
+
+// replace img links in commentHTML with route to proxy, base64 encoded original link
+func (p Image) replace(commentHTML string, imgs []string) string {
+	for _, img := range imgs {
+		encodedImgURL := base64.URLEncoding.EncodeToString([]byte(img))
+		resImgURL := p.RemarkURL + p.RoutePath + "?src=" + encodedImgURL
+		commentHTML = strings.Replace(commentHTML, img, resImgURL, -1)
 	}
 
-	return p.replace(commentHTML, imgs)
+	return commentHTML
 }
 
 // Handler returns http handler respond to proxied request
 func (p Image) Handler(w http.ResponseWriter, r *http.Request) {
-
-	if !p.Enabled {
-		http.Error(w, "proxy disabled", http.StatusNotImplemented)
+	if !p.HTTP2HTTPS && !p.CacheExternal {
```

Comment:
yeah, that might be a solution. I don't think it's critical to fix it right now, how about I create an issue to have it fixed later in order to avoid holding this PR any longer?

Reply:
agree

Reply:
@smaant could you please create an issue for that?

Reply:
Resolving in favor of #614.